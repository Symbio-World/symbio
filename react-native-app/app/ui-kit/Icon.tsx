import * as React from 'react'

import { SvgProps } from 'react-native-svg'
import camera from '../../assets/svg/camera.svg'
import cross from '../../assets/svg/cross.svg'
import disk from '../../assets/svg/disk.svg'
import flashlight from '../../assets/svg/flashlight.svg'
import historyicon from '../../assets/svg/history.svg'
import fatalError from '../../assets/svg/fogg-fatal-error.svg'

type Props = SvgProps & {
  icon: Icons
}

export enum Icons {
  CAMERA = 'CAMERA',
  CROSS = 'CROSS',
  DISK = 'DISK',
  FLASHLIGHT = 'FLASHLIGHT',
  HISTORYICON = 'HISTORYICON',
  FATAL_ERROR = 'FATAL_ERROR',
}

const IconsMap = {
  [Icons.CAMERA]: camera,
  [Icons.CROSS]: cross,
  [Icons.DISK]: disk,
  [Icons.FLASHLIGHT]: flashlight,
  [Icons.HISTORYICON]: historyicon,
  [Icons.FATAL_ERROR]: fatalError,
}

export const Icon: React.FC<Props> = ({ icon, ...otherProps }) => {
  const IconComponent = IconsMap[icon]
  if (!IconComponent) throw new Error('No icon found')
  return <IconComponent {...otherProps} />
}
