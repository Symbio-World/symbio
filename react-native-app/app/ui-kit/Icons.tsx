import * as React from 'react'
import MaterialIcon from 'react-native-vector-icons/MaterialIcons';

export 

