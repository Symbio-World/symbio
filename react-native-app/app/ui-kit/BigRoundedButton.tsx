import * as React from 'react'
import { TouchableOpacity, Text, View } from 'react-native'
import { t } from 'react-native-tailwindcss'

import { Icon, Icons } from './Icon'

type Props = {
  title: string
  icon: React.ReactNode
  onPress?: () => void
  variant?: 'outline' | 'fill'
}

const ButtonStyle = {
  fill: [t.bgPrimary],
  outline: [t.bgWhite, t.border2, t.borderPrimary],
}

const TitleStyle = {
  fill: [t.textWhite],
  outline: [t.textPrimary],
}

export const BigRoundedButton: React.FC<Props> = ({
  title,
  icon,
  onPress = () => {},
  variant = 'fill',
}) => {
  const buttonStyle = ButtonStyle[variant]
  const titleStyle = TitleStyle[variant]

  return (
    <TouchableOpacity onPress={onPress}>
      <View
        style={[
          ...buttonStyle,
          t.w24,
          t.h24,
          t.roundedFull,
          t.flexCol,
          t.itemsCenter,
          t.justifyCenter,
        ]}
      >
        <View style={[t.w6, t.h6, t.mB2]}>{icon}</View>
        <View>
          <Text style={[t.textCenter, t.textBase, t.fontBold, ...titleStyle]}>
            {title}
          </Text>
        </View>
      </View>
    </TouchableOpacity>
  )
}
