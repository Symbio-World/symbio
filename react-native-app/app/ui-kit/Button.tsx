import * as React from 'react'
import { TouchableOpacity, Text } from 'react-native'
import { t } from 'react-native-tailwindcss'

import { Icon, IconsEnum } from './Icon'

export enum ButtonStyle {
  FILL,
  OUTLINE,
  TRANSPARENT,
}

type Props = {
  onPress?: () => void
  title?: string
  icon?: IconsEnum
  variant?: ButtonStyle
}

const ButtonBaseStyle = {
  [ButtonStyle.FILL]: [t.bgPrimary],
  [ButtonStyle.OUTLINE]: [t.bgWhite, t.border2, t.borderPrimary],
  [ButtonStyle.TRANSPARENT]: [t.bgTransparent],
}

const InnerTitleBaseStyle = {
  [ButtonStyle.FILL]: [t.textWhite],
  [ButtonStyle.OUTLINE]: [t.textPrimary],
  [ButtonStyle.TRANSPARENT]: [t.textPrimary],
}

export const Button: React.FC<Props> = ({
  onPress = () => {},
  title = '',
  variant = ButtonStyle.FILL,
  icon = null,
}) => {
  const buttonStyle = ButtonBaseStyle[variant]
  const innerStyle = [InnerTitleBaseStyle[variant]]
  return (
    <TouchableOpacity
      style={[
        t.p4,
        t.itemsCenter,
        t.justifyCenter,
        t.flex,
        t.flexRow,
        t.rounded,
        ...buttonStyle,
      ]}
      onPress={onPress}
    >
      {icon ? <Icon style={[t.mR2]} iconType={icon} /> : null}
      <Text style={[t.textCenter, t.text2xl, t.textWhite, ...innerStyle]}>
        {title}
      </Text>
    </TouchableOpacity>
  )
}
