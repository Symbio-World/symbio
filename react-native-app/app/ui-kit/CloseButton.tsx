import * as React from 'react'

import { RoundedButton } from './RoundedButton'
import { ButtonStyle } from './Button'
import { Icons } from './Icon'

type CloseButtonProps = {
  onClose?: () => void
}

export const CloseButton: React.FC<CloseButtonProps> = ({ onClose }) => (
  <RoundedButton
    onPress={onClose}
    outerTitle="close"
    variant={ButtonStyle.OUTLINE}
    icon={Icons.CROSS}
  />
)
