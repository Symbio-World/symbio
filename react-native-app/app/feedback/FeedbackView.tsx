import * as React from 'react'
import { View, Text, TextInput, Keyboard } from 'react-native'
import { t } from 'react-native-tailwindcss'
import Toast from 'react-native-simple-toast'
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import { ModalActionsView } from '../ui-kit/ModalActionsView'
import { CloseButton } from '../ui-kit/CloseButton'
import { RoundedButton } from '../ui-kit/RoundedButton'
import { Icons } from '../ui-kit/Icon'

type Props = {
  onSubmit?: (email: string) => void
}

export const FeedbackView: React.FC<Props> = ({ onSubmit = () => {} }) => {
  const [text, setText] = React.useState<string>('')
  const handleSubmit = () => {
    onSubmit(text)
    Keyboard.dismiss()
    Toast.show('Your email was saved successfully')
  }
  return (
    <KeyboardAwareScrollView
      contentContainerStyle={[t.flex1, t.mT6, t.mB3, t.mX5]}
      enableOnAndroid
      enableAutomaticScroll
    >
      <Text style={[t.text2xl, t.mB6, t.textCenter]}>Help us improve!</Text>
      <Text style={[t.textBase, t.mB4]}>
        Hi there, thanks for using Symbio! We want to make it better, so weâ€™re
        seeking your help to find out what you need. Will you be willing to take
        part in a short customer interview? Let us know by entering your email
        below so we can contact you. ðŸ™‚
      </Text>
      <Text style={[t.textBase, t.mB3]}>Your email</Text>
      <View style={[t.flex1]}>
        <TextInput
          onChangeText={setText}
          onSubmitEditing={handleSubmit}
          value={text}
          returnKeyType="send"
          autoFocus
          style={[
            t.border,
            t.borderGray500,
            t.h12,
            t.roundedLg,
            t.p3,
            t.textBlack,
          ]}
        />
      </View>
      <ModalActionsView>
        <CloseButton onClose={() => {}} />
        <RoundedButton
          onPress={handleSubmit}
          outerTitle="send"
          icon={Icons.SEND}
        />
      </ModalActionsView>
    </KeyboardAwareScrollView>
  )
}
