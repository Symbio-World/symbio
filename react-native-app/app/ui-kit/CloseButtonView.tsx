import * as React from 'react'
import { t } from 'react-native-tailwindcss'
import { CloseButton } from './CloseButton'
import { View } from 'react-native'

type CloseButtonViewProps = {
  onPress?: () => void
}

export const CloseButtonView: React.FC<CloseButtonViewProps> = ({
  onPress,
}) => (
  <>
    <View style={[t.h40]} />
    <View style={[t.absolute, t.bottom0, t.wFull, t.itemsCenter, t.pB5]}>
      <CloseButton onPress={onPress} />
    </View>
  </>
)
