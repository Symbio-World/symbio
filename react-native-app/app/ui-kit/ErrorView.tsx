import * as React from 'react'
import { View, Text } from 'react-native'
import { t } from 'react-native-tailwindcss'
import { Icon, Icons } from './Icon'
import { CloseButtonView } from './CloseButtonView'

type Props = {
  error?: unknown
  onPress?: () => void
}
export const ErrorView: React.FC<Props> = ({ onPress = () => {} }) => {
  return (
    <View style={[t.flex1]}>
      <Icon icon={Icons.FATAL_ERROR} />
      <Text style={[t.textCenter, t.textBlack, t.text5xl, t.mB10]}>
        OOOOPS!
      </Text>
      <Text style={[t.textCenter, t.textBlack, t.text3xl, t.mX12]}>
        Something went wrong... You can tell us what happened or try again later
      </Text>
      <CloseButtonView onPress={onPress} />
    </View>
  )
}
