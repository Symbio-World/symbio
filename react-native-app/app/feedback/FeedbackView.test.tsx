import * as React from 'react'
import { TextInput, Keyboard } from 'react-native'
import { Button } from '../ui-kit/Button'
import { render, fireEvent } from 'react-native-testing-library'

import { FeedbackView } from './FeedbackView'

jest.mock('react-native/Libraries/Components/Keyboard/Keyboard', () => ({
  addListener: jest.fn(),
  dismiss: jest.fn(),
}))

describe('FeedbackView', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<FeedbackView />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('triggers onSubmit', () => {
    const feedback = 'feedback'
    const handleSubmit = jest.fn()
    const { getByType } = render(<FeedbackView onSubmit={handleSubmit} />)
    fireEvent.changeText(getByType(TextInput), feedback)
    fireEvent.press(getByType(Button))
    expect(handleSubmit).toHaveBeenCalledWith(feedback)
  })

  it('triggers keyboard dissmiss', () => {
    const feedback = 'feedback'
    const handleSubmit = jest.fn()
    const { getByType } = render(<FeedbackView onSubmit={handleSubmit} />)
    fireEvent.changeText(getByType(TextInput), feedback)
    fireEvent.press(getByType(Button))
    expect(Keyboard.dismiss).toHaveBeenCalled()
  })
})
