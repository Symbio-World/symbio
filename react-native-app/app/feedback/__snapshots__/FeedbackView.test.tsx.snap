// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackView renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "marginVertical": 64,
      },
      Object {
        "marginHorizontal": 20,
      },
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <Text
      style={
        Array [
          Object {
            "fontSize": 24,
          },
          Object {
            "fontWeight": "bold",
          },
          Object {
            "marginVertical": 16,
          },
        ]
      }
    >
      ðŸ’¬
      
    </Text>
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoFocus={true}
        multiline={true}
        numberOfLines={4}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        rejectResponderTermination={true}
        returnKeyType="send"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#48bb78",
          "borderRadius": 4,
          "display": "flex",
          "justifyContent": "center",
          "opacity": 1,
          "padding": 16,
          "textAlignVertical": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "fontSize": 24,
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        Submit feedback
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
